src/lib.rs

```diff
@@ -19,47 +19,68 @@ fn take_guess() -> char {
         .expect("More than one character was guessed.");
 }
 
-fn run_game_loop(config: &mut Config, input: char) -> GameState {
-    if !config.secret_word.contains(input) {
-        return GameState::LOST;
+struct PlayerState {
+    wrong_guesses: u32,
+}
+
+fn run_game_loop(config: &mut Config, guess: char, player_state: &mut PlayerState) -> GameState {
+    if !config.secret_word.contains(guess) {
+        player_state.wrong_guesses += 1;
+
+        if player_state.wrong_guesses > config.lives {
+            return GameState::LOST;
+        } else {
+            println!(
+                "WRONG... lives remaining {}",
+                config.lives - player_state.wrong_guesses
+            );
+            return GameState::PLAYING;
+        }
     }
 
+    update_correctly_guessed_letters(config, guess);
+
+    println!("CORRECT... status: {}", config.correctly_guessed_letters);
+
+    if config.correctly_guessed_letters == config.secret_word {
+        return GameState::WON;
+    }
+
+    // return GameState::PLAYING;
+    return GameState::PLAYING;
+}
+
+fn update_correctly_guessed_letters(config: &mut Config, guess: char) {
     // check for hits of guess in solution
     let hit_idxs: Vec<usize> = config
         .secret_word
         .char_indices()
-        .filter_map(|(i, c)| if c == input { Some(i) } else { None })
+        .filter_map(|(i, c)| if c == guess { Some(i) } else { None })
         .collect();
 
     // Fill in the correctly guessed letters in the user facing message
     for hit_idx in hit_idxs {
         config
             .correctly_guessed_letters
-            .replace_range(hit_idx..hit_idx + 1, &input.to_string());
-    }
-
-    println!("status: {}", config.correctly_guessed_letters);
-
-    if config.correctly_guessed_letters == config.secret_word {
-        return GameState::WON;
+            .replace_range(hit_idx..hit_idx + 1, &guess.to_string());
     }
-
-    return GameState::PLAYING;
 }
 
 pub fn run(config: &mut Config) -> Result<(), Box<dyn Error>> {
-    println!("{}", config.correctly_guessed_letters);
```

Comment:
[nit]: Now that we got rid of this print the first guess is "blind" in the sense that you have no information about the length of the word you are guessing. I would not propose to address this in the current PR but it may make sense to think about how we want to handle the reporting to the user.